### ISSUE TITLE
Add speed cameras (including average speed ranges)

### ISSUE THREAD

**Issue by Talkless** [rinigus/osmscout-server#399]
Hi,

I've noticed that once I've enabled offline maps & routing in uNav app, calculated route no longer have speed cameras displayed on it.

uNav developer states that offline routing API does not provide that info: https://github.com/costales/unav/issues/51

Could it be done in osmscout setup? I'm not sure how all of this works, what limitations it might have.

I imagine, once route is calculated, route legs could be checked from the OSM-based databas if:

- There's `highway=speed_camera`node  on that route leg (way). `maxspeed=...` key should be set, that could be shown on map/warning.
- There's relation with `type=enforcement`, with `enforcement=maxspeed` or `enforcement=average_speed`, where route leg (way) is part of that relation as `segment`, or at least `from` ant `to` nodes are on that calculated route. This relation must have `highway=speed_camera` node, but it might be _near_ the way, not part of the way. In other words if `highway=speed_camera` is not on the way itself, there must be enforcement relation.

see https://wiki.openstreetmap.org/wiki/Relation:enforcement 
and https://wiki.openstreetmap.org/wiki/Key:maxspeed

Showing speed cameras are important especially for people who drive in new unfamiliar places, there you can overspeed simply accidentally...

**Comment by rinigus** [comment]
Indeed, OSM Scout Server does not provide speed cameras info. I actually don't know what are the legal aspects of showing such data. Quick search found info from TomTom (https://help.tomtom.com/hc/en-us/articles/360013899980-TomTom-Speed-Camera-service-and-the-law). 

There is a general data regarding current speed limit - unrelated to cameras. But that is available through map matching at DBus, which wouldn't work on Ubuntu Touch (works on Sailfish and should on other mobile Linux distros).

**Comment by Talkless** [comment]
> I actually don't know what are the legal aspects of showing such data

OSMAnd allows to remove speed camera data. Probably there could be separate data set (download) that user could decide to download or not, depending on is legality.

**Comment by rinigus** [comment]
I see. That is probably a way to do it. I will keep this issue open, just in case if someone wants to work on it. However, before any active work, it would make sense to agree on exported API.
